File: db4o.net/Db4objects.Db4o/Db4objects.Db4o/Internal/Activation/LegacyFixedUpdateDepth.cs
/* Copyright (C) 2004 - 2011  Versant Inc.  http://www.db4o.com */

using Db4objects.Db4o.Internal;
using Db4objects.Db4o.Internal.Activation;

namespace Db4objects.Db4o.Internal.Activation
{
	public class LegacyFixedUpdateDepth : FixedUpdateDepth
	{
		public LegacyFixedUpdateDepth(int depth) : base(depth)
		{
		}

		public override bool CanSkip(ObjectReference @ref)
		{
			return false;
		}

		protected override FixedUpdateDepth ForDepth(int depth)
		{
			return new Db4objects.Db4o.Internal.Activation.LegacyFixedUpdateDepth(depth);
		}
	}
}

File: db4oj/core/src/com/db4o/internal/activation/LegacyFixedUpdateDepth.java
package com.db4o.internal.activation;

import com.db4o.internal.*;

public class LegacyFixedUpdateDepth extends FixedUpdateDepth {

	public LegacyFixedUpdateDepth(int depth) {
		super(depth);
	}

	public boolean canSkip(ObjectReference ref) {
		return false;
	}

	@Override
	protected FixedUpdateDepth forDepth(int depth) {
		return new LegacyFixedUpdateDepth(depth);
	}

}
